keps/sig-cli/3659-kubectl-apply-prune/README.md

```diff
@@ -627,128 +544,171 @@ In future, we may define additional “index” mechanisms here to further optim
+When pruning with `--applyset`, kubectl will delete objects that are labeled as part of the ApplySet of objects, but are not in the list of objects being applied. Dry-run will be supported initially, and `kubectl diff --prune --applyset=id` support will be added before beta.
 
 The `--prune` flag will continue to be required for all pruning operations to ensure communication of intent for this destructive feature. The `--applyset` flag has no meaning on its own and specifying it without `--prune` is an error. We will not support any of the scoping flags used by the previous pruning feature, that is, `--prune-allowlist`, `-l/--selector` and `--all`. These are considered conflicting pruning "modes", and specifying them alongside `--applyset` will fail flag validation. Our goal is to support the existing safe workflows, not the full permutations of all flags. The allowlist function in particular should be redundant with our improved discovery. What meaning the label selector flag would have if allowed is unclear, and we will need to collaborate with kubectl users to understand their true intent if there is demand for compatibility with that flag.
 
-The `--namespace` flag will be required when using any namespaced parent, including the default Secret. Because that flag throws a mismatch error when the set contains resources with heterogeneous namespaces, this limits the scope of ApplySet-based pruning in kubectl specifically beyond what the spec proposed strictly requires. Specifically, in kubectl, ApplySets spanning multiple namespaces MUST use a cluster-scoped parent object. We believe this limitation is reasonable and encourages best practices, but we could consider relaxing this position (e.g. using the ApplySet-in-input option described above) based on user feedback. When applicable, kubectl will ensure that all "visited" namespaces (as defined in the current operational code) are named by the sum of the parent's own namespace (if any) and the `applyset.k8s.io/additional-namespaces` annotation.
-
-We will detect “overlapping” ApplySets where objects already have a different ApplySet label, and initially treat this
-an error. During implementation of the alpha we will explore to what extent we can
-optimize this overlap discovery, particularly in conjunction with
-server-side-apply which does not require an object read before applying.
-A richer apply tooling than kubectl does will likely establish watches
-on the objects before applying them, to monitor object health and status.
-However, this is out of scope for kubectl and thus we will likely have to
-optimize differently for kubectl.  In the worst case, we will have to fetch
-the objects before applying (with a set of label-filtered LIST requests),
-we will explore to what extent that can be amortized over other kubectl
```

Comment:
Being more familiar with the requests made during CSA and SSA, I don't think there's any way to amortize it in SSA.